### PR TITLE
Added feature to get changelogs between given 2 tags.

### PR DESCRIPTION
**Pull request by jatinkksharma**
Now we can generate changelogs between 2 given tags.

### CODE REVIEW

File: src/run.ts [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
-import { rsort } from 'semver'
+import { rsort } from 'semver';
 
 export async function run(): Promise<void> {
     await ensureGit();
     await ensureGitRepository();
-
-    const lines = await getLogLines(Config.sourceTag);
+    const sortedTags = await getVersions();
+    if (!Config.currentTag || Config.currentTag === undefined) {
```
Comment (ghost) [inline src/run.ts]: `!Config.currentTag` includes `Config.currentTag === undefined`. We can remove the later condition.

File: src/run.ts [hunk 2]
```diff
@@ -1,17 +1,44 @@
...
+        console.error('Please provide current tag');
+        process.exit(1);
+    }
+    const currentIndex = await sortedTags.indexOf(Config.currentTag.substr(1));
+    if(currentIndex === -1) {
+        console.error('Invalid current tag');
```
Comment (ghost) [inline src/run.ts]: How about `Current tag doesn't exist, tag: <tag>`? Invalid is not really the correct terminology here.

File: src/run.ts [hunk 3]
```diff
@@ -1,17 +1,44 @@
...
+        process.exit(1);
+    }
+    if (Config.previousTag) {
```
Comment (ghost) [inline src/run.ts:30]: Add an empty line between two `if` blocks. Also, how about integrating a linter?

File: src/run.ts [hunk 4]
```diff
@@ -1,17 +1,44 @@
...
+        const previousIndex = await sortedTags.indexOf(Config.previousTag.substr(1));
+        if (previousIndex === -1) {
+            console.error('Invalid previous tag');
+            process.exit(1);
+        } else if (previousIndex <= currentIndex) {
+            console.error('Previous tag should be less than current tag');
+            process.exit(1);
+        }
+    }
+    if(!Config.previousTag) {
```
Comment (ghost) [inline src/run.ts]: I think we can simply use `else` here.

File: src/run.ts (hunk 1 above)
Comment (ghost) [inline src/run.ts]: Lets see if we can modularize this logic?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/run.ts (hunk 1 above)
Comment (jatinkksharma) [reply src/run.ts]: Done

File: src/run.ts (hunk 2 above)
Comment (jatinkksharma) [reply src/run.ts]: Done

File: src/run.ts (hunk 4 above)
Comment (jatinkksharma) [reply src/run.ts]: Done

File: src/run.ts (hunk 1 above)
Comment (jatinkksharma) [reply src/run.ts]: Done

File: src/run.ts (hunk 3 above)
Comment (jatinjtg) [reply src/run.ts]: Done

Comment (jatinkksharma) [reply src/run.ts:30]: I have integrated tslint .It dosen't allows whitespace

Comment (ghost) [reply src/run.ts:30]: You must be having a space in that blank line. Maybe that's causing the error?

**Review by ghost — APPROVED** [review]

File: src/run.ts (hunk 3 above)
Comment (jatinkksharma) [reply src/run.ts:30]: Okay.

**Pull request merged**